### ISSUE TITLE
codebase is impenetrable because of billions of source directories

### ISSUE THREAD

**Issue by chessai** [andrewthad/sockets#32]
title is hyperbole

would cabal 3's support for multiple public libraries help this?

if not, could the layout be documented? it's currently hard to navigate

**Comment by andrewthad** [comment]
I don't think that multiple public libraries would be useful here. What I need to do is not just document the layout but also rename a few of the internal modules while I'm at it.

**Comment by chessai** [comment]
okay, thanks
